### DISCUSSION TITLE
能否增加一键将输入和答案保存到对应该的.in和.out文件中

### DISCUSSION THREAD

**Discussion by lpstudio520** [cpeditor/cpeditor-cn#42] [想法]
能否增加一键将输入和答案保存到对应该的.in和.out文件中，这样便于出题者可以直接当作数据制作工具用。

**Comment by ouuan** [comment]
> 能否增加一键将输入和答案保存到对应该的.in和.out文件中

现在设置里可以开启自动保存所有测例，只不过这个功能设计是为了方便重启时恢复数据，而不是导出测例。另有从文件批量导入测例的功能、保存单个 in/out/ans 到任意文件的功能、一键把所有输出复制到答案。

> 这样便于出题者可以直接当作数据制作工具用

我认为 CP Editor 不适合用来造数据。如果你想这样造数据的话，我认为你应该是在纯手动造数据，这应该只适用于入门难度的题目。如果你是想自动跑 std 生成输出，一个简单的 shell/bat/Python 脚本就可以做到。可以参考 https://oi-wiki.org/tools/testlib/generator/ 、<https://oi-wiki.org/contest/problemsetting/#造数据>、<https://oi-wiki.org/tools/polygon/>。只不过我才意识到 OI Wiki 这里没有提及怎么自动跑 std，可以去加一下（

**Reply by lpstudio520** [reply]
@就是入门级的题目，现在全国绝大部分小学都要求信息课开展c++课程，老师要给学生出点题，有时造数据很是头疼。
